cognite_toolkit/_cdf_tk/utils/interactive_select.py

```diff
@@ -366,6 +369,82 @@ def _select_external_ids(self, select_filter: CanvasFilter) -> list[str]:
         return selected_canvases or []
 
 
+@dataclass
+class ChartFilter:
+    visibility: Visibility | None = None
+    owned_by: Literal["user"] | None = None
+    select_all: bool = False
+
+
+class InteractiveChartSelect:
+    opening_choices: ClassVar[list[questionary.Choice]] = [
+        questionary.Choice(title="All public Charts", value=ChartFilter(visibility="PUBLIC", select_all=True)),
+        questionary.Choice(title="Selected public Charts", value=ChartFilter(visibility="PUBLIC", select_all=False)),
+        questionary.Choice(title="All owned by given user", value=ChartFilter(owned_by="user", select_all=True)),
+        questionary.Choice(title="Selected owned by given user", value=ChartFilter(owned_by="user", select_all=False)),
+    ]
+
+    def __init__(self, client: ToolkitClient) -> None:
+        self.client = client
+
+    def select_external_ids(self) -> list[str]:
+        select_filter = self._select_filter()
+        return self._select_external_ids(select_filter)
+
+    @classmethod
+    def _select_filter(cls) -> ChartFilter:
+        user_response = questionary.select(
+            "Which Charts do you want to select?",
+            choices=cls.opening_choices,
+        ).ask()
+        if user_response is None:
+            raise ToolkitValueError("No Chart selection made. Aborting.")
+        return user_response
+
+    def _select_external_ids(self, select_filter: ChartFilter) -> list[str]:
+        available_charts = self.client.charts.list(visibility=(select_filter.visibility or "PUBLIC"))
+        if select_filter.select_all and select_filter.owned_by is None:
+            return [chart.external_id for chart in available_charts]
+        users = self.client.iam.user_profiles.list(limit=-1)
+        display_name_by_user_identifier = {user.user_identifier: user.display_name or "missing" for user in users}
```

Comment:
Seeing how this is used (dict.get() with a default value equal to the key) wouldn't it degrade better if the user_identifier without a display name was simply not included as a key in the map rather than mapped to "missing"?